lib/image-transform.js

```
@@ -96,9 +96,20 @@ module.exports = class ImageTransform {
 	 * Set the device-pixel ratio of the transformed image.
 	 * @param {(String|Number)} [value=1] - The DPR.
 	 */
-	setDpr(value) {
+	setDpr(value, width, height) {
 		const errorMessage = 'Image DPR must be a positive whole number';
 		this.dpr = ImageTransform.sanitizeNumericValue(value, errorMessage);
+		if(this.dpr > 1) {
+			if (this.height) {
+				this.setHeight(Math.floor(height / this.dpr));
+			}
+			if (this.width) {
+				this.setWidth(Math.floor(width / this.dpr));
+			}
+			if(!this.width && !this.height) {
+				throw new Error (errorMessage);
```

I think should throw a specific error. Something like:
```
Image DPR can only be set when the transform has an explicit width or height
```